src/java/org/apache/cassandra/tools/nodetool/TableHistograms.java

```
@@ -57,11 +57,15 @@ else if (args.size() == 1)
         {
             checkArgument(false, "tablehistograms requires keyspace and table name arguments");
         }
-
-        // calculate percentile of row size and column count
-        long[] estimatedPartitionSize = (long[]) probe.getColumnFamilyMetric(keyspace, table, "EstimatedPartitionSizeHistogram");
-        long[] estimatedColumnCount = (long[]) probe.getColumnFamilyMetric(keyspace, table, "EstimatedColumnCountHistogram");
-
+        long[] estimatedPartitionSize;
+        long[] estimatedColumnCount;
+        try {
+            // calculate percentile of row size and column count
+            estimatedPartitionSize = (long[]) probe.getColumnFamilyMetric(keyspace, table, "EstimatedPartitionSizeHistogram");
+            estimatedColumnCount = (long[]) probe.getColumnFamilyMetric(keyspace, table, "EstimatedColumnCountHistogram");
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Unknown table: " + table + " in keyspace: " + keyspace);
```

Is this error message correct, though? AFAICT, we can fail if either the table or keyspace is missing, right? So the error message should be "Unknown table <table name> or keyspace <keyspace name>"?

---

you right, it's probably better